src/common/config/dir_list.cpp

```diff
@@ -33,25 +33,44 @@ void ParsedPath::parse(const PathName& path)
 {
 	clear();
 
-	if (path.length() == 1) {
-		add(path);
-		return;
-	}
-
 	PathName oldpath = path;
+	int toSkip = 0;
 	do {
 		PathName newpath, elem;
 		PathUtils::splitLastComponent(newpath, elem, oldpath);
 		oldpath = newpath;
+		if (elem.isEmpty() && !oldpath.isEmpty()) // Skip double dir separator
+		{
+			continue;
+		}
+		if (elem == ".") // Skip current dir reference
+		{
+			continue;
+		}
+		if (elem == PathUtils::up_dir_link) // skip next up dir
```

Comment:
Does FB still run (compiles for) in Win 9X based kernels? I do remember in this OS "..." (or more dots) was valid part of paths, meaning "..\.." and so. Is this compatible? Or is the old code (2.X) compatible with this?